osu.Game/Overlays/Mods/ModSection.cs

```diff
@@ -128,15 +128,16 @@ public void DeselectTypes(IEnumerable<Type> modTypes, bool immediate = false)
         /// Select one or more mods in this section and deselects all other ones.
         /// </summary>
         /// <param name="modTypes">The types of <see cref="Mod"/>s which should be selected.</param>
-        public void SelectTypes(IEnumerable<Type> modTypes)
+        /// <param name="deselect">Whether to deselect mods not present in <paramref name="modTypes"/></param>
+        public void SelectTypes(IEnumerable<Type> modTypes, bool deselect = true)
```

Comment:
Having a `SelectTypes` method with a `deselect` parameter is quite dubious, especially given that its xmldoc shows the name is misleading. It should at worst be named `deselectOthers`, and at best not exist at all.

Have you looked at enforcing this via specifying `IncompatibleMods`?

Reply:
Actually the method I came up with doesn't look fine anymore to me for the sole reason that I discovered it was working fine because it made the assumption that `ModAutoplay` and `ModCinema` were inside a MultiMod (that's the case for the 4 default rulesets, tested this by moving the `Autoplay` and `Cinema` mod implementations out of the MultiMod for the osu! ruleset). So i'll probably remove the whole method.

So I may now be more in favour of adding a check inside `trySelectMod` to ensure that the selected auto mod isn't incompatible with the already selected mods (by checking against `IncompatibleMods` property + a `IsSubclassOfCheck` (would love to hear your opinion on this).

I may give this a shot tomorrow but can't say for sure.